### ISSUE TITLE
[3.2.x] SoftDelete Behavior Does not Implement Snapshots

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#13173]
The `\Phalcon\Mvc\Behavior\SoftDelete` class doesn't implement snapshots. The reason this is important is because the model mimics the `save()` behavior by aborting the current `delete()` operation to perform a silent update of the field.

### Expected and Actual Behavior

```php
class Users extends Model
{
    const DELETED = "D";
    const NOT_DELETED = "N";

    public $id;
    public $name;
    public $status;

    public function initialize()
    {
        $this->addBehavior(new SoftDelete([
           "field" => "status",
           "value" => Users::DELETED
        ]));
    }
}
```

```php
$user = Users::findFirst();
$user->delete();

$user->getUpdatedFields();
// Expected:  ['status']
// Actual: []
```

**Comment by Jurigag** [comment]
Well you wrote wrong reason - the reason is beacause model on which is save done is cloned from original model, in this model snapshot is sure updated. But in old one not.

**Comment by virgofx** [comment]
I realize the update happens in a clone; the title is still correct. The original model should contain the updated snapshot fields... which is the core issue.

**Comment by Jurigag** [comment]
Yea yea, i just mean the reason why there is no updated snapshot is other you suggest.

**Comment by sergeyklay** [comment]
Fixed in the `3.3.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
